zio-http/jvm/src/main/scala/zio/http/netty/client/NettyConnectionPool.scala

```diff
@@ -142,23 +142,31 @@ private[netty] object NettyConnectionPool {
   }
 
   /**
-   * Refreshes the idle timeout handler on the channel pipeline.
+   * Attempts to refresh the idle timeout handler on the channel pipeline.
    * @return
    *   true if the handler was successfully refreshed prior to the channel being
-   *   closed
+   *   closed and the channel is still open.
+   *
+   * true if the `timeout` is None and the channel is still open.
+   *
+   * false of the channel is closed.
    */
   private def refreshIdleTimeoutHandler(
     channel: JChannel,
-    timeout: Duration,
-  ): Boolean = {
-    channel
-      .pipeline()
-      .replace(
-        Names.ReadTimeoutHandler,
-        Names.ReadTimeoutHandler,
-        new ReadTimeoutHandler(timeout.toMillis, TimeUnit.MILLISECONDS),
-      )
-    channel.isOpen
+    timeout: Option[Duration],
+  )(implicit trace: Trace): Task[Boolean] = {
+    ZIO.attempt {
+      timeout.map { timeout =>
+        channel
+          .pipeline()
+          .replace(
+            Names.ReadTimeoutHandler,
+            Names.ReadTimeoutHandler,
+            new ReadTimeoutHandler(timeout.toMillis, TimeUnit.MILLISECONDS),
+          )
+      }
+    }.when(channel.isOpen)
+      .as(channel.isOpen)
```

Comment:
This "breaks" the current logic for retrying channels from the pool until we succeed. I think a better solution would be to keep the existing code as is and wrap it in a `try-catch` block so that we don't fail requests when this happens

```scala
  private def refreshIdleTimeoutHandler(
    channel: JChannel,
    timeout: Duration,
  ): Boolean = {
  try {
    channel
      .pipeline()
      .replace(
        Names.ReadTimeoutHandler,
        Names.ReadTimeoutHandler,
        new ReadTimeoutHandler(timeout.toMillis, TimeUnit.MILLISECONDS),
      )
    channel.isOpen
  } catch {
    case _: NoSuchElementException => false
  }
```